python/lsst/sims/catalogs/measures/instance/InstanceCatalog.py

```diff
@@ -273,6 +273,35 @@ def __init__(self, db_obj, obs_metadata=None, column_outputs=None,
         #the columns in self._column_origins (we only want to do that once)
         self._column_origins_switch = True
 
+        #now we will create and populate a list containing the names of
+        #all of the columns which this InstanceCatalog can return.
+        #Note: this needs to happen before self._check_requirements()
+        #is called in case any getters depend on the contents of
+        #_all_available_columns.  That way, self._check_requirements()
+        #can verify that the getter will run the way it is actually
+        #being called.
+        self._all_available_columns = []
+
+        for name in self.db_obj.columnMap.keys():
+            self._all_available_columns.append(name)
+
+        for name in self._compound_column_names:
+            self._all_available_columns.append(name)
+
+        for name in self._compound_columns:
+            self._all_available_columns.append(name)
+
+        for name in dir(self):
+            if name[:4]=='get_':
+                columnName = name[4:]
+                if columnName not in self._all_available_columns:
+                    self._all_available_columns.append(columnName)
+            elif name[:8]=='default_':
+                columnName = name[8:]
+                if columnName not in self._all_available_columns:
+                    self._all_available_columns.append(columnName)
+
```

Comment:
I am not sure if this makes all items in the list self._all_available_columns unique? Could items from columnMap.keys(), default_cols also be used in _compound_column_names? D

Does this matter?